### PR TITLE
Arafat Khan's guest post

### PR DESCRIPTION
**Pull request by jdorfman**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205978769636858
  - https://app.asana.com/0/0/1206073908510706

### CODE REVIEW

**Comment by jdorfman** [comment]
🔴  Don't merge yet 🔴

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (arafatkatze) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:230]:
```suggestion
I asked a question here that touches 11 different files, each of which would easily exceed the context window of GPT 4 (Never mind the copy-pasting hassle). It sifts through all the files and shows me how they are interconnected and what calls touch which parts. This to me, is the most important feature of Cody: the ability to answer questions that are deeply tangled in the complexity of the codebase. It helps me answer questions that I simply can’t find through a simple copy-paste because the real problem is sometimes _I don’t even know where to look_ when I am dealing with very large codebases.
```

Comment (arafatkatze) [reply content/blogposts/2023/guest-post-arafat-khan-php-python.md:230]: image

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (arafatkatze) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:220]:
```suggestion
All these features, such as _autocomplete, explanation, code fixes, and testing,_ made my job as a programmer a lot easier. I was able to build the building blocks of my code, and then, inch by inch, I was getting closer to completing the entire project. As I was building more and more of this codebase, a question that crossed my mind was:
```

Comment (arafatkatze) [reply content/blogposts/2023/guest-post-arafat-khan-php-python.md:220]: image

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (arafatkatze) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:222]:
```suggestion
## Why use Cody when you have Chat GPT-4?
```

Ideally its best to update this segment of this blog to say that now Cody will also support GPT 4.5 Turbo which does have the very high context window coz some people(Myself included) would be such a huge fan of it that we are more than happy to pay for it and even use our own keys. Folks already made the knowledge of its existence public in [this blog](https://about.sourcegraph.com/blog/cody-vscode-0-16-release) so perhaps its best to change the argument to. Instead of copy pasting in ChatGPT(which is what I used to do before) now you got it all in Cody.

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (arafatkatze) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:72]:
```suggestion
  id="N-9QUjpv1qY"
```

Comment (arafatkatze) [reply content/blogposts/2023/guest-post-arafat-khan-php-python.md:72]: The video here is this one -> https://www.youtube.com/watch?v=N-9QUjpv1qY

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (arafatkatze) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:217]:
```suggestion
            <div>
                <YouTube title="Autocomplete with Cody" id="DgwvhRW1Cbc" />
            </div>
```

Comment (arafatkatze) [reply content/blogposts/2023/guest-post-arafat-khan-php-python.md:217]: Adding the autocomplete example video

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (arafatkatze) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:71]:
```suggestion
  title="Cody select and explain" 
```

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (arafatkatze) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:249]:
```suggestion
    To anyone embarking on a similar journey or facing the challenge of diving into unfamiliar territory, I would say that AI tools in 2023 are like the internet in the late 90s. You can deny its existence and impact all you wish, but there is very soon going to be a future that’s so deeply entwined with AI that you wouldn’t be able to think of a life without it. I strongly recommend people try out AI coding assistants. There’s quite a few out there, but Cody is the best one.
```

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (arafatkatze) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:224]: ❤️

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (arafatkatze) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:230]:
```suggestion
I asked a question here that touches 11 different files, each of which would easily exceed the context window of ChatGPT (Never mind the copy-pasting hassle). It sifts through all the files and shows me how they are interconnected and what calls touch which parts. This to me, is the most important feature of Cody: the ability to answer questions that are deeply tangled in the complexity of the codebase. It helps me answer questions that I simply can’t find through a simple copy-paste because the real problem is sometimes _I don’t even know where to look_ when I am dealing with very large codebases. And here's the best part: Cody is soon going to support Gpt-4.5-turbo(So all the Open AI fans still get to stay with their favourite models)
```

You can add something along these lines if you wanna also mention those things although ultimately its your call.

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (arafatkatze) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:224]:
```suggestion
I like the speed and convenience of Cody but I can’t quite tell why using the Vscode extension is so much better than using Chatgpt if I just wanted to ask code questions. This is where the real magic of Cody comes in. Cody can answer questions about your entire codebase and sift through all the different parts to find the most relevant information pertaining to your question. It uses the [Claude 100K model](https://www.anthropic.com/index/100k-context-windows) (soon to be the [200K model](https://www.anthropic.com/index/claude-2-1)) so it has the capacity to fit in so much more context than my copy-pasting hacks ever could. Take, for example, this question
```

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:8]: When tasked with creating a printing API in PHP despite having zero experience, Arafat Khan leveraged Cody to transform what felt like an Everest climb into a swift 3 day project that resulted in a fully functioning API.

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:10]: "building-a-printing-api-in-php-with-cody"

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:20]: A few weeks ago, a professional endeavor landed on my desk creating a printing API. => A few weeks ago, a professional endeavor landed on my desk: creating a printing API.

With a functional Python API version already in my arsenal => With an already functional Python API version in my arsenal

To add to the challenge, I imposed a time constraint on myself so now I had to learn a new language and build a full-blown project that I could share with the world. => To add to the challenge, I imposed a time constraint on myself. So now I had to learn a new language and build a full-blown project that I could share with the world.

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:24]: remove "behind some classes."

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:35]: While I knew that I knew nothing about PHP I also knew that I had help that I could rely on. I started with the only two things that I had at the time: => While I knew that I knew nothing about PHP, I also knew that I had help that I could rely on. I started with the only two things that I had at the time:

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:38]: An AI-assisted coding editor from Sourcegraph => A code AI assistant from Sourcegraph

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:40]: Vscode => VS Code

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:56]: internet => Internet

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:60]: setting => settings

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:77]: starting point of => starting point for

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:182]: This is a very decent scaffolding for the entire Auth class we have the private/public variables, most of the auth method already implemented, and the outline of the send method. This is a great starting point, I already learned so much about PHP syntax just from this, and now I can build things on top of it. Brick by brick, by selecting segments of my Python code, I can understand them and then create them in PHP, too. I kept doing the iterative work of selecting different parts of my Python code and asking more questions on it to piece it all together to make my PHP code. =>

This is a very decent scaffold for the entire Auth class. We have the private/public variables, most of the auth method already implemented, and the outline of the send method. This is a great starting point, I already learned so much about PHP syntax just from this, and now I can build things on top of it. Brick by brick, by selecting segments of my Python code, I can understand them and then re-create them in PHP, too. I kept doing the iterative work of selecting different parts of my Python code and asking more questions on it to piece it all together to make my PHP code.

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:185]: Obviously, we are dealing with LLMs here, so nothing is perfect. Time and again, LLMs would hallucinate and give me imperfect responses, but in those cases, I would have to use my programming background and some common sense to tweak the code. I would also try my best to write as much code myself coz that teaches you the intricacies of the syntax and the common mistakes that people tend to make, and it’s generally a great learning experience. =>

Obviously, we are dealing with LLMs here, so nothing is perfect. Occasionally, Cody would hallucinate and give me an imperfect response, and in those cases, I would have to use my programming background and some common sense to tweak the code. I would also try my best to write as much code myself because that teaches you the intricacies of the syntax and the common mistakes that people tend to make, and it’s generally a great learning experience.

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:188]: remove ", and you are good to go".

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:208]: 3. Generate near-perfect code

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:220]: remove " a classic"

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:226]: I like the speed and convenience of Cody but I can’t quite tell why using the Vscode extension is so much better than using ChatGPT if I just wanted to ask code questions. This is where the real magic of Cody comes in. =>

I like the speed and convenience of Cody, as well as the ability to use it directly in VS Code, but there is another major reason why I prefer Cody.

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:232]: I simply can’t find through a simple copy-paste because the real problem is sometimes _I don’t even know where to look_ when I am dealing with very large codebases. =>

 I simply can’t find through a simple copy-paste because the real problem is that sometimes _I don’t even know where to look_ when I am dealing with very large codebases.

Comment (kukicado) [reply content/blogposts/2023/guest-post-arafat-khan-php-python.md:232]: Gpt-4.5-turbo => GPT-4.5-Turbo

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:235]: Looks like the title is a line below the #

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:239]: So for now, that’s a good outlook of how Cody helped me. It’s made me faster and improved my dev workflow so much that I don’t think I will be coding without it in the future. The daunting Everest climb of learning PHP began to feel more like a stroll in the park. Every obstacle I encountered was promptly addressed, either by directly consulting Cody or by utilizing the resources and code suggestions it provided.

=> 

Cody has made me faster and improved my dev workflow so much that I don’t think I will be coding without it in the future. The daunting Everest climb of learning PHP began to feel more like a stroll in the park. Every obstacle I encountered was promptly addressed, either by directly consulting Cody or by utilizing the resources and code suggestions it provided.

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:251]: internet => Internet

wouldn't => won't

AI coding assistants => Cody

remove "there's quite a few out there..."

File: content/blogposts/2023/guest-post-arafat-khan-php-python.md
Comment (kukicado) [inline content/blogposts/2023/guest-post-arafat-khan-php-python.md:256]: Can we have this go to the Cody landing page?

**Review by kukicado — DISMISSED** [review]
Awesome post! I added feedback on improving the copy, but the messaging is really great.

**Comment by arafatkatze** [comment]
@kukicado Thanks a lot for such a detailed feedback. One tiny recommendation: It might be very nice to use edit suggestions([See point 6](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-comments-to-a-pull-request)). It looks like this 

image

And makes it super easy to merge suggestions(It even works for multiple lines). [Example](https://github.com/sourcegraph/about/pull/6586#discussion_r1401514278)


I added the changes in this [PR](https://github.com/sourcegraph/about/pull/6608). Let me know if you have any further suggestions and thanks again for such a detailed feedback.

**Comment by jdorfman** [comment]
Closing this since #6608 was merged

**Pull request merged**
